velox/connectors/hive/storage_adapters/hdfs/tests/CMakeLists.txt

```diff
@@ -32,3 +31,27 @@ target_link_libraries(
 
 target_compile_options(velox_hdfs_file_test
                        PRIVATE -Wno-deprecated-declarations)
+
+add_executable(velox_hdfs_insert_test InsertIntoHdfsTest.cpp
+                                      HdfsMiniCluster.cpp HdfsUtilTest.cpp)
+
+add_test(velox_hdfs_insert_test velox_hdfs_insert_test)
+
+target_link_libraries(
+  velox_hdfs_insert_test
+  velox_file
+  velox_hdfs
+  velox_core
+  velox_exec_test_lib
+  velox_hive_connector
+  velox_dwio_common_exception
+  velox_exec
+  GTest::gtest
+  GTest::gtest_main
+  GTest::gmock)
+
+target_compile_options(velox_hdfs_insert_test
+                       PRIVATE -Wno-deprecated-declarations)
+
+set_tests_properties(velox_hdfs_insert_test PROPERTIES DEPENDS
```

Comment:
@majetideepak To resolve the `port is in use` issue when `velox_hdfs_insert_test `and `velox_hdfs_file_test` are executed concurrently, configure  to start `velox_hdfs_file_test `only after `velox_hdfs_insert_test `has completed.

Reply:
Can you use the PortUtil https://github.com/facebookincubator/velox/blob/c3c93bf607435e6ae75379ca3b34e364f05a83ce/velox/exec/tests/utils/PortUtil.h#L4 to fix this problem?

Reply:
@majetideepak I've  resolved the `port in use` issue by using the method from `PortUtil.h` to find a free port, which allowed the `velox_hdfs_insert_test ` to pass successfully. However, I'm now facing an exception when executing the `velox_hdfs_file_test`. The exception shows that the file `/test_file.txt` is not found, even though it has been created as specified in the test setup [here](https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/storage_adapters/hdfs/tests/HdfsFileSystemTest.cpp#L49). Interestingly, this test passes on my local machine. Do you have any suggestions on what might be causing this issue in the Velox CI environment? Thanks.

```
Running main() from /__w/velox/velox/_build/release/_deps/gtest-src/googletest/src/gtest_main.cc
[==========] Running 21 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 20 tests from HdfsFileSystemTest
WARNING: HADOOP_PREFIX has been replaced by HADOOP_HOME. Using value of HADOOP_PREFIX.
WARNING: HADOOP_PREFIX has been replaced by HADOOP_HOME. Using value of HADOOP_PREFIX.
put: File /test_file.txt._COPYING_ could only be written to 0 of the 1 minReplication nodes. There are 1 datanode(s) running and 1 node(s) are excluded in this operation.
[ RUN      ] HdfsFileSystemTest.read
2024-09-29 06:31:36,411 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
WARNING: Logging before InitGoogleLogging() is written to STDERR
E20240929 06:31:36.913120 63709 Exceptions.h:66] Line: /__w/velox/velox/velox/connectors/hive/storage_adapters/hdfs/HdfsReadFile.cpp:36, Function:HdfsReadFile, Expression:  Unable to get file path info for file: hdfs://localhost:7878/test_file.txt. got error: FileNotFoundException: Path hdfs://localhost:7878/test_file.txt does not exist., Source: RUNTIME, ErrorCode: FILE_NOT_FOUND
unknown file: Failure
C++ exception with description "Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: FILE_NOT_FOUND
Reason: Unable to get file path info for file: hdfs://localhost:7878/test_file.txt. got error: FileNotFoundException: Path hdfs://localhost:7878/test_file.txt does not exist.
Retriable: False
Function: HdfsReadFile
File: /__w/velox/velox/velox/connectors/hive/storage_adapters/hdfs/HdfsReadFile.cpp
Line: 36
Stack trace:
# 0  _ZN8facebook5velox7process10StackTraceC1Ei
# 1  _ZN8facebook5velox14VeloxExceptionC2EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bNS1_4TypeES7_
# 2  _ZN8facebook5velox6detail14veloxCheckFailINS0_17VeloxRuntimeErrorERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEvRKNS1_18VeloxCheckFailArgsET0_
# 3  _ZN8facebook5velox12HdfsReadFileC2EPNS0_11filesystems5arrow2io8internal11LibHdfsShimEP13hdfs_internalSt17basic_string_viewIcSt11char_traitsIcEE
# 4  _ZN28HdfsFileSystemTest_read_Test8TestBodyEv
# 5  _ZN7testing8internal35HandleExceptionsInMethodIfSupportedINS_4TestEvEET0_PT_MS4_FS3_vEPKc
# 6  _ZN7testing4Test3RunEv
# 7  _ZN7testing8TestInfo3RunEv
# 8  _ZN7testing9TestSuite3RunEv
# 9  _ZN7testing8internal12UnitTestImpl11RunAllTestsEv
# 10 _ZN7testing8UnitTest3RunEv
# 11 main
# 12 __libc_start_call_main
# 13 __libc_start_main
# 14 _start
" thrown in the test body.
[  FAILED  ] HdfsFileSystemTest.read (992 ms)
```

Reply:
@JkSelf I see the following message. Does it have any implication of writing the file?
`put: File /test_file.txt._COPYING_ could only be written to 0 of the 1 minReplication nodes.`

Reply:
@majetideepak Good catch. It appears that the issue is caused by the DataNode port being occupied. The [jar ](https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/storage_adapters/hdfs/tests/HdfsMiniCluster.h#L29) we are using does not provide an option to configure the DataNode port [here](https://github.com/apache/hadoop/blob/release-3.3.0-RC0/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapreduce/MiniHadoopClusterManager.java#L89-L119). We currently have two solutions:

1. Modify the Hadoop jar to add an option for configuring the DataNode port.
2. Use the previous approach and run `velox_hdfs_insert_test  `after `velox_hdfs_file_test `done.

The first solution has a higher maintenance cost. I prefer the second solution. What's your suggestions? Thanks.

Reply:
I agree, the second solution is easier. Please add a comment about this to the CMakeFile for future reference. Thanks.